### ISSUE TITLE
runMigrationSilent stomps on the encoding of stderr handle

### ISSUE THREAD

**Issue by joeyh** [yesodweb/persistent#474]
This transcript shows a really strange side effect, and surely it's a bug this happens:

```
*Database.Fsck Database.Handle> hSetEncoding stderr utf16
*Database.Fsck Database.Handle> hGetEncoding stderr
Just UTF-16
*Database.Fsck Database.Handle> initDb "/tmp/testdb" $ void $ runMigrationSilent  migrateFsck 
*Database.Fsck Database.Handle> hGetEncoding stderr
Just UTF-8
```

stdout's encoding is not effected, only stderr.

I'm using persistent-2.1.6 and persistent-sqlite-2.1.4.2, with ghc 7.8.4.

In case it matters somehow, here's the TH code that generated the migration:

```
share [mkPersist sqlSettings, mkMigrate "migrateFsck"] [persistLowerCase|
  Fscked
    key SKey
    UniqueKey key
  |]
newtype SKey = SKey String
    deriving (Show, Read)
```

**Comment by gregwebs** [comment]
Can you send this report upstream?

``` haskell
import System.IO.Silently (hSilence)
runMigrationSilent m = liftBaseOp_ (hSilence [stderr]) $ runMigration' m True
```

**Comment by sol** [comment]
In general, I think it would be more robust to not rely on `silently` to control verbosity.

**Comment by parsonsmatt** [comment]
Fixed by #971
